src/main/java/spoon/reflect/meta/impl/RoleHandlerHelper.java

```diff
@@ -71,4 +78,47 @@ public static RoleHandler getOptionalRoleHandler(Class<? extends CtElement> targ
 		}
 		return null;
 	}
+
+	/**
+	 * @param targetClass a Close whose handlers we are looking for
+	 * @return all RoleHandlers available for the `targetClass`
+	 */
+	public static List<RoleHandler> getRoleHandlers(Class<? extends CtElement> targetClass) {
+		List<RoleHandler> handlers = roleHandlersByClass.get(targetClass);
+		if (handlers == null) {
+			List<RoleHandler> modifiableHandlers = new ArrayList<>();
+			for (CtRole role : CtRole.values()) {
+				RoleHandler roleHandler = getOptionalRoleHandler(targetClass, role);
+				if (roleHandler != null) {
+					modifiableHandlers.add(roleHandler);
+				}
+			}
+			handlers = Collections.unmodifiableList(modifiableHandlers);
+			roleHandlersByClass.put(targetClass, handlers);
+		}
+		return handlers;
+	}
+
+	/**
+	 * @param consumer is called for each RoleHandler of SpoonModel
+	 */
+	public static void forEachRoleHandler(Consumer<RoleHandler> consumer) {
+		for (List<RoleHandler> list : roleHandlers) {
+			for (RoleHandler roleHandler : list) {
+				consumer.accept(roleHandler);
+			}
+		}
+	}
+
+	public static RoleHandler getParentRoleHandler(CtElement element) {
```

Comment:
rename to getRoleHandler?

add API documentation?

thanks!

Reply:
> rename to getRoleHandler?

are you sure? I think that there should be something about `parent` in the method name, because it returns role of this element in parent.

> api doc.
I forgot that ... It compiles without documentation well :-) I will add it

Reply:
> because it returns role of this element in parent.

a role is always with respect to the parent, right?

Reply:
in case of this method yes.